### DISCUSSION TITLE
Does the BSL license in this project allows the use of Uptrace on production-grade applications?

### DISCUSSION THREAD

**Discussion by jhhom** [uptrace/uptrace#72] [Q&A] [answered]
Hi, would like to confirm about one question that I have about the license on this tool.

Under the license, I know that I cannot resell this tool as my product to others, but can I use this tool for monitoring and instrumenting my web applications that are in business and making money?

Because based on this [page](https://mariadb.com/bsl-faq-adopting/), it says that:

_The BSL is structured to allow free and open usage for many use cases, and only requires a commercial license by those who make production use of the software, which is typically indicative of an environment that is delivering significant value to a business._

"Only requires a commercial license by those who make production use of the software", does this means I'm not allowed to self-host Uptrace to monitor my production software without paying any price to the Uptrace organization? Instead I will need to use the cloud version of Uptrace?

Thanks.

**Comment by vmihailenco** [comment, marked as answer]
The sentence you are quoting is a bit vague indeed, but you **can** use Uptrace to monitor commercial/business applications without any limitations and provide your employees access to the app.

The BSL license only dis-allows re-selling and/or re-packaging Uptrace, e.g. creating another cloud service using the Uptrace code or just selling the product under your name.

I will update README with relevant Q/A.

**Reply by jhhom** [reply]
Thanks for the answer, that clears it up for me.
